src/components/submission/diff-item.vue

```diff
@@ -0,0 +1,267 @@
+<!--
+Copyright 2021 ODK Central Developers
+See the NOTICE file at the top-level directory of this distribution and at
+https://github.com/getodk/central-frontend/blob/master/NOTICE.
+
+This file is part of ODK Central. It is subject to the license terms in
+the LICENSE file found in the top-level directory of this distribution and at
+https://www.apache.org/licenses/LICENSE-2.0. No part of ODK Central,
+including this file, may be copied, modified, propagated, or distributed
+except according to the terms contained in the LICENSE file.
+-->
+<template>
+  <div class="submission-diff-item" :class="nestedClass">
+    <div v-if="visiblePath.length > 0" class="full-path">
+      <span v-for="(field, index) in visiblePath" :key="index">
+        <template v-if="Array.isArray(field)">
+          <span>{{ field[0] }}</span>
+          <span class="field-counter"> #</span><span>{{ field[1] + 1 }}</span>
+        </template>
+        <template v-else>
+          <span>{{ field }}</span>
+        </template>
+        <template v-if="index < visiblePath.length - 1">
+          <span class="field-separator"> &rsaquo; </span>
+        </template>
+      </span>
+    </div>
+    <div class="diff-details">
+      <template v-if="isAtomicChange">
+        <div class="field-name" :title="fieldName">
+          {{ fieldName }}
+        </div>
+        <div class="old-to-new">
+          <span v-if="entry.old" class="data-old">
+            <template v-if="isBinary">
+              <a :href="binaryHref(entry.old)" :title="entry.old">{{ formatData(entry.old) }}</a>
+            </template>
+            <span v-else :title="entry.old">{{ formatData(entry.old) }}</span>
+          </span>
```

Comment:
If you prefer, you could also put the `title` attribute on the `.data-old` and `.data-new` elements rather than their children. One nice result of that is that the `v-else` case could remain a `<template>` rather than rendering another `<span>`.